src/app/penalties/penalties.ts

```diff
@@ -119,27 +125,67 @@ export class PenaltiesComponent implements OnInit {
   }
 
   onPageClicked(pageNumber) {
-    if (pageNumber < 1) {
-      pageNumber = 1;
-    } else if (pageNumber > this.pages.length) {
-      pageNumber = this.pages.length;
-    }
-    if (pageNumber !== this.activePage) {
-      this.activePage = pageNumber;
-      this.getPenalties().publish().connect();
+    if (!this.isValidPage(pageNumber)) {
+      return;
     }
+
+    this.activePage = pageNumber;
+    this.getPenalties()
+      .publish()
+      .connect();
+  }
+
+  isValidPage(page: number): boolean {
+    let isPositive,
+        isLessThatMax;
+
+    isPositive = 1 <= page;
+    isLessThatMax = page <= _.last(this.pages);
+
+    return isPositive && isLessThatMax;
+  }
+
+  createPaginationInfo(res) {
```

Comment:
Both `isValidPage` and `createPaginationInfo` can go to `PenaltiesPagination` component, not sure why are they part of `Penalties` component....